src/rgw/rgw_sal_dbstore.h

```diff
@@ -259,6 +260,130 @@ class DBNotification : public Notification {
     }
   };
 
+  /*
+   * For multipart upload, below is the process flow -
+   *
+   * MultipartUpload::Init - create head object of the original object (if already doesn't
+   *                    exist) & meta obj (src_obj_name + "." + upload_id)
+   *                     [ Meta object stores all the parts upload info]
```

Comment:
does creating this 'head object' during Init mean that clients can try to access it already, before the parts are uploaded?

i'm also confused about the 'meta obj' - how is this represented in the database? would it be visible to clients as the object name "src_obj_name.upload_id"?

in rados, we don't create the final head object until MultipartComplete to make sure the partial object isn't ever visible to clients

Reply:
Sorry. I forgot to upload the updated comments. As mentioned the PR description, head_object is created or overwrittern only during MultipartUpload::Complete .

I have taken meta obj reference from sal_rados which creates object of type "RGWMPObj" to store upload_info and mp parts info (used for listing / abort)

Reply:
meta_obj is deleted as soon as the upload is complete (RGWCompleteMultipart::execute->meta_obj->delete()). 
I haven't checked during upload. But it may be filtered with 'Object.ns != RGW_OBJ_NS_MULTIPART' during listing.